### PR TITLE
chore: Added initial package setup and OpenAPI client

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

**Comment by Raalsky** [comment]
> I think that we're missing license headers in the files (and perhaps a relevant pre-commit hook to insert those?)

AFAIK, it's only required if you have multiple licenses across the codebase. In `neptune-client` we had `pynvml` and that was the main reason. I don't think it's necessary here.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - id: trailing-whitespace
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        args: [--settings-path, pyproject.toml]
+  - repo: https://github.com/psf/black
+    rev: 24.2.0
+    hooks:
+      - id: black
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply .pre-commit-config.yaml:16]: Pytorch Lightning does have `ruff` + `black` so probably let's stay with both of them :wink:

File: dev_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-e .
+
+# dev
+openapi-python-client>=0.20,<0.21
+pre-commit
+mypy
+ruff
+isort
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply dev_requirements.txt:8]: Looks like we already have `I` selected in ruff settings. I had to add `isort` in post-hooks but probably due to ruff missing our pyproject settings

[the remark this answers is not in the document]
Comment (Raalsky) [reply dev_requirements.txt:8]: Nah, `ruff` doesn't cover `force_grid_wrap` and the `isort`-compatibility is still under development so we have to stay with `isort` as well :<.

**Review by AleksanderWWW — APPROVED** [review]

**Pull request merged**